02_knight_and_princess/robert_gaisin/main.cpp

```diff
@@ -69,13 +84,14 @@ class Graph {
   VertexId next_vertex_id() { return num_of_vrt_++; }
   EdgeId next_edge_id() { return num_of_edg_++; }
 
-  bool is_vertex(const VertexId& vertex_id) const;
+  bool has_vertex(const VertexId& vertex_id) const;
+  bool has_edge_id(const EdgeId& edge_id);
```

Comment:
1. `const`?
2. Ну почему такое непостоянство? Разве ты сам не видишь, 2 функции, выполняющие одинаковую логику, написаны одна после другой, но называются по-разному. Не бывает хороших, удобных интерфейсов, где нет постоянства. Будь внимательней.
```suggestion
  bool has_vertex(const VertexId& vertex_id) const;
  bool has_edge(const EdgeId& edge_id) const;
```
или
```suggestion
  bool has_vertex_id(const VertexId& vertex_id) const;
  bool has_edge_id(const EdgeId& edge_id) const;
```
Первый вариант лучше, потому что граф состоит из вершин и граней, а не из их `id`